### ISSUE TITLE
CMake does not work on macOS due to rt requirement

### ISSUE THREAD

**Issue by nlohmann** [aras-p/ClangBuildAnalyzer#69]
The `rt` library is unconditionally added as requirement in file `CMakeLists.txt`:

```cmake
target_link_libraries(ClangBuildAnalyzer -lrt -lpthread)
```

This library is not present on macOS, and hence the build fails.

When I remove the `-lrt` entry, building succeeds.

The line above should be changed like:

```cmake
if (APPLE)
  target_link_libraries(ClangBuildAnalyzer -lpthread)
else()
  target_link_libraries(ClangBuildAnalyzer -lrt -lpthread)
endif()
```

I can make a PR if requested.

**Comment by mathbunnyru** [comment]
There's already that fixes this problem: https://github.com/aras-p/ClangBuildAnalyzer/pull/62

@aras-p may I kindly ask you to take a look, please?

**Comment by nlohmann** [comment]
Yes, #62 would fix this.

**Comment by aras-p** [comment]
Ok, blindly merged #62 (GitHub Actions are seemingly failing everywhere with internal errors today). 🤞
